pkg/util/mount/mount_linux.go

```diff
@@ -115,6 +117,36 @@ func doMount(mounterPath string, mountCmd string, source string, target string,
 		mountCmd = mounterPath
 	}
 
+	if _, err := os.Stat(systemdRunPath); err == nil {
```

Comment:
kubelet in container runs NsEnterMounter and that runs systemd-run in the host mount namespace.